components/brave_shields/browser/domain_block_navigation_throttle.cc

```diff
@@ -0,0 +1,175 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/domain_block_navigation_throttle.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/metrics/histogram_macros.h"
+#include "base/task/post_task.h"
+#include "base/threading/thread_task_runner_handle.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/domain_block_controller_client.h"
+#include "brave/components/brave_shields/browser/domain_block_page.h"
+#include "brave/components/brave_shields/browser/domain_block_tab_storage.h"
+#include "brave/components/brave_shields/common/features.h"
+#include "components/prefs/pref_service.h"
+#include "components/security_interstitials/content/security_interstitial_tab_helper.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_user_data.h"
+#include "net/base/net_errors.h"
+
+namespace {
+
+bool ShouldBlockDomainOnTaskRunner(
+    brave_shields::AdBlockService* ad_block_service,
+    const GURL& url) {
+  SCOPED_UMA_HISTOGRAM_TIMER("Brave.DomainBlock.ShouldBlock");
+  if (!ad_block_service->IsInitialized())
```

Comment:
this isn't safe and `IsInitialized` should have a thread dcheck because initialized_ is set in `StartRegionalServices` which runs on the UI thread. You can pass this in to the method as a bool instead

Reply:
actually instead of passing this in as a bool you can just return early and never call this method

Reply:
moved the IsInitialized check to WillStartRequest which runs on UI thread and has a thread check